api/src/repositories/survey-repository.ts

```diff
@@ -1100,4 +1101,188 @@ export class SurveyRepository extends BaseRepository {
+
+    const sqlStatement = SQL`
+      SELECT
+        fn.first_nations_id as id,
+        fn.name as first_nations_name
+      FROM
+        survey_first_nation_partnership sfnp
+      LEFT OUTER JOIN
+        first_nations fn
+      ON
+      sfnp.first_nations_id = fn.first_nations_id
+      WHERE
+      sfnp.survey_id = ${surveyId}
+      GROUP BY
+        fn.first_nations_id,
+        fn.name;
+    `;
+
+    const response = await this.connection.query(sqlStatement.text, sqlStatement.values);
+
+    const result = response?.rows;
+
+    if (!result) {
+      throw new ApiExecuteSQLError('Failed to get survey Indigenous Partnerships data', [
+        'SurveyRepository->getIndigenousPartnershipsBySurveyId',
+        'rows was null or undefined, expected rows != null'
+      ]);
+    }
+
+    return result;
+  }
+
+  /**
+   * @TODO jsdoc
+   *
+   * @TODO should probably simplify this method to use knex? And accept [] as a valid response? ...
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<GetPartnershipsData['stakeholder_partnerships'][]>}
+   * @memberof SurveyRepository
+   */
+  async getStakeholderPartnershipsBySurveyId(
+    surveyId: number
+  ): Promise<GetPartnershipsData['stakeholder_partnerships'][]> {
+    defaultLog.debug({ label: 'getStakeholderPartnershipsBySurveyId', surveyId });
+
+    const sqlStatement = SQL`
+      SELECT
+        name as partnership_name
+      FROM
+        survey_stakeholder_partnership
+      WHERE
+        survey_id = ${surveyId};
+    `;
+
+    const response = await this.connection.query(sqlStatement.text, sqlStatement.values);
+
+    const result = response?.rows;
+
+    if (!result) {
+      throw new ApiExecuteSQLError('Failed to get survey Stakeholder Partnerships data', [
+        'SurveyRepository->getStakeholderPartnershipsBySurveyId',
+        'rows was null or undefined, expected rows != null'
+      ]);
+    }
+
+    return result;
+  }
+
+  /**
+   * Inserts indegenous partnership records for the given survey
+   *
+   * @param {number[]} firstNationsIds
+   * @param {number} surveyId
+   * @return {*}  {Promise<{ survey_first_nation_partnership_id: number }[]>}
+   * @memberof SurveyRepository
+   */
+  async insertIndigenousPartnerships(
+    firstNationsIds: number[],
+    surveyId: number
+  ): Promise<{ survey_first_nation_partnership_id: number }[]> {
+    defaultLog.debug({ label: 'insertIndigenousPartnerships', firstNationsIds, surveyId });
+    const queryBuilder = getKnex()
+      .table('survey_first_nation_partnership')
+      .insert(
+        firstNationsIds.map((firstNationsId: number) => ({
+          first_nations_id: firstNationsId,
+          survey_id: surveyId
+        }))
+      )
+      .returning('survey_first_nation_partnership_id');
+
+    const response = await this.connection.knex<{ survey_first_nation_partnership_id: number }>(queryBuilder);
+
+    if (!response || response.rowCount === 0) {
```

Comment:
response will always be defined, so could drop the initial `!response` check

Reply:
Lots of our old code makes that check because early on I had the types wrong (was under the impression that response could be undefined, but turns out it cant be, otherwise an exception would be thrown by node-pg).

Reply:
Will remove these checks!